canopy_alloc.F90

```diff
@@ -70,5 +70,51 @@ SUBROUTINE canopy_alloc
         if(.not.allocated(rjcf_3d))         allocate(rjcf_3d(nlon,nlat,modlays))
     end if
 
+!-------------------------------------------------------------------------------
+! Allocate arrays for Canopy Biogenic Emissions Outputs
+!-------------------------------------------------------------------------------
+
+    if (ifcanbio) then
+        write(*,*)  'Canopy biogenic emissions option selected'
+        write(*,*)  '-------------------------------'
+        if(.not.allocated(emi_isop))         allocate(emi_isop(nlat*nlon,modlays))
```

Comment:
I think a lot of the code could be made more extendable and maintainable if we used internally a single 4-d emi array instead, with species dimension.

Reply:
@zmoon I had thought about that, and tend to agree with you.  I thought even the species desired could be controlled though the NL, but didn't want to go down that path right away.  I wanted just something to work initially.

So, can you make a new branch off of this and start to make these changes to a 4-d "emi_bio" array and we can compare how it looks?

Reply:
@zmoon Thanks!  Makes sense to work on it here, but let me know if we want to merge this large PR first and then branch off, its up to you.

Reply:
> we want to merge this large PR first and then branch off, its up to you.

Yeahhh I think this is a better idea. We can come back to it if we want add to add a bunch more species at some point or something.